kani-compiler/src/codegen_cprover_gotoc/codegen/foreign_function.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! This module implements foreign function handling.
+//!
+//! Kani currently only support CBMC built-in functions that are declared in the `cprover_bindings`
+//! crate, and allocation functions defined in `kani_lib.c`.
+//!
+//! All other functions will be replaced by an unimplemented check, due to current issues with
+//! linking and usability unless unstable C-FFI support is enabled.
+use std::collections::HashSet;
+
+use crate::codegen_cprover_gotoc::codegen::PropertyClass;
+use crate::codegen_cprover_gotoc::GotocCtx;
+use crate::kani_middle;
+use cbmc::goto_program::{Expr, Location, Stmt, Symbol, Type};
+use cbmc::{InternString, InternedString};
+use kani_queries::UserInput;
+use lazy_static::lazy_static;
+use rustc_middle::ty::Instance;
+use rustc_target::abi::call::Conv;
+use tracing::{debug, trace};
+
+lazy_static! {
+    static ref RUST_ALLOC_FNS: HashSet<InternedString> = {
+        HashSet::from([
+            "__rust_alloc".into(),
+            "__rust_alloc_zeroed".into(),
+            "__rust_dealloc".into(),
+            "__rust_realloc".into(),
```

Comment:
Where does this list come from?

Reply:
I am adding a comment to describe this in the next version. Basically these are declared in the [Rust alloc module](https://stdrs.dev/nightly/x86_64-unknown-linux-gnu/alloc/alloc/index.html), but they are defined today in [kani_lib.c](https://github.com/model-checking/kani/blob/main/library/kani/kani_lib.c).